sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -647,12 +647,18 @@ private class CompletedOperation : IOperation
         {
             private readonly OperationState _operationState;
 
-            public CompletedOperation(OperationState operationState)
+            private readonly Uri _startRequestUri;
+
+            public CompletedOperation(OperationState operationState, Uri startRequestUri)
             {
                 _operationState = operationState;
+                _startRequestUri = startRequestUri;
             }
 
             public ValueTask<OperationState> UpdateStateAsync(bool async, CancellationToken cancellationToken) => new(_operationState);
+
+            public RehydrationToken GetRehydrationToken() =>
+                NextLinkOperationImplementation.GetRehydrationToken(RequestMethod.Put, _startRequestUri, _startRequestUri.AbsoluteUri, "None", null, OperationFinalStateVia.OriginalUri.ToString());
```

Comment:
The logic here will vary based on the type of method the LRO is wrapping, the method will also need to vary based on the actual method, we should be able to pull the values from the instance of nextlinkoperation that we are currently in.

Reply:
Updated! Also added some tests for CompletedOperation.